tests/functional/aws-node-sdk/test/multipleBackend/acl/aclAwsVersioning.js

```diff
@@ -0,0 +1,228 @@
+const assert = require('assert');
+const async = require('async');
+const withV4 = require('../../support/withV4');
+const BucketUtility = require('../../../lib/utility/bucket-util');
+const constants = require('../../../../../../constants');
+const {
+    awsLocation,
+    enableVersioning,
+    putNullVersionsToAws,
+    putVersionsToAws,
+    getAndAssertResult,
+    describeSkipIfNotMultiple,
+} = require('../utils');
+
+const someBody = 'testbody';
+const bucket = 'buckettestmultiplebackendaclsawsversioning';
+
+class _AccessControlPolicy {
+    constructor(params) {
+        this.Owner = {};
+        this.Owner.ID = params.ownerID;
+        if (params.ownerDisplayName) {
+            this.Owner.DisplayName = params.ownerDisplayName;
+        }
+        this.Grants = [];
+    }
+    addGrantee(type, value, permission, displayName) {
+        const grant = {
+            Grantee: {
+                Type: type,
+            },
+            Permission: permission,
+        };
+        if (displayName) {
+            grant.Grantee.DisplayName = displayName;
+        }
+        if (type === 'AmazonCustomerByEmail') {
+            grant.Grantee.EmailAddress = value;
+        } else if (type === 'CanonicalUser') {
+            grant.Grantee.ID = value;
+        } else if (type === 'Group') {
+            grant.Grantee.URI = value;
+        }
+        this.Grants.push(grant);
+    }
+}
+
+const ownerParams = {
+    ownerID: '79a59df900b949e55d96a1e698fbacedfd6e09d98eacf8f8d5218e7cd47ef2be',
+    ownerDisplayName: 'Bart',
+};
+const testAcp = new _AccessControlPolicy(ownerParams);
+testAcp.addGrantee('Group', constants.publicId, 'READ');
+
+function putObjectAcl(s3, key, versionId, acp, cb) {
+    s3.putObjectAcl({ Bucket: bucket, Key: key, AccessControlPolicy: acp,
+        VersionId: versionId }, err => {
+        assert.strictEqual(err, null, 'Expected success ' +
+            `putting object acl, got error ${err}`);
+        cb();
+    });
+}
+
+function putObjectAndAcl(s3, key, body, acp, cb) {
+    s3.putObject({ Bucket: bucket, Key: key, Body: body,
+    Metadata: { 'scal-location-constraint': awsLocation } },
+    (err, putData) => {
+        assert.strictEqual(err, null, 'Expected success ' +
+            `putting object, got error ${err}`);
+        putObjectAcl(s3, key, putData.VersionId, acp, () =>
+            cb(null, putData.VersionId));
+    });
+}
+
+function putVersionsWithAclToAws(s3, key, data, acps, cb) {
```

Comment:
please do jsdoc for this one. unclear that data is an array of objects and acps are array of different access control policies.